### PR TITLE
add esp32 p4 stub and target code

### PR DESCRIPTION
**Pull request by brianignacio5**
Add ESP32 P4 support.

Test steps:
```
npm i
npm run build
cd examples/typescript
npm i
npm run dev
```

Open a browser and go to `http://localhost:1234` to launch esptool-js website.

### CODE REVIEW

**Comment by brianignacio5** [comment]
PTAL @RushikeshPatange

**Review by adwait-esp — APPROVED** [review]

**Comment by adwait-esp** [comment]
LGTM

File: src/esploader.ts [hunk 1]
```diff
@@ -155,8 +155,7 @@ async function magic2Chip(magic: number): Promise<ROM | null> {
       const { ESP8266ROM } = await import("./targets/esp8266.js");
       return new ESP8266ROM();
     }
-    case 0x0addbad0:
-    case 0x0: {
```
Comment (adwait-esp) [inline src/esploader.ts]: The previous change seemed right - Magic value for p4 chip seems to be - 0x0addbad0

Comment (brianignacio5) [reply src/esploader.ts]: Adding them now

Comment (igrr) [reply src/esploader.ts]: @adwait-esp which chip revision do you have? (You can check using esptool.py)

Comment (brianignacio5) [reply src/esploader.ts]: Those values are listed in current esptool master. I've tested with Function EV Board but magic value was `0x7039ad9`

**Review by adwait-esp — APPROVED** [review]

**Pull request merged**